src/Microsoft.Health.Core/Features/Health/HealthStatusReason.cs

```diff
@@ -5,8 +5,13 @@
 
 namespace Microsoft.Health.Core.Features.Health;
 
-public enum ExternalHealthReason
+/// <summary>
+/// The reason why the service is in its current health state. In order of severity
```

Comment:
It is important to code implementation. In HealthReportExtensions.GetHighestSeverityReason, we compare the enum value to decide which reason has the highest severity. 

Right now only when the HealthStatus is Degraded will we need to do this, since we have a special "Reason" for CMK and we will have one for IDP and probably other things in the future. If multiple things are wrong, we have to decide which one is the "worst" so we have a single reason to report to the customer.

I added some more comments to hopefully make this more clear, so when we are adding to this in the future people will uphold the severity order. What do you think?